roles/custom/matrix-user-verification-service/tasks/setup_uninstall.yml

```diff
@@ -0,0 +1,35 @@
+---
+
+- name: Check existence of matrix-user-verification-service service
+  stat:
+    path: "{{ matrix_systemd_path }}/{{ matrix_user_verification_service_systemd_service_name }}"
+  register: matrix_user_verification_service_service_stat
+
+- name: Ensure matrix-user-verification-service is stopped
+  service:
+    name: "{{ matrix_user_verification_service_systemd_service_basename }}"
+    state: stopped
+    daemon_reload: yes
+  register: stopping_result
+  when: "matrix_user_verification_service_service_stat.stat.exists|bool"
+
+- name: Ensure matrix-user-verification-service.service doesn't exist
+  file:
+    path: "{{ matrix_systemd_path }}/{{ matrix_user_verification_service_systemd_service_name }}"
+    state: absent
+  when: "matrix_user_verification_service_service_stat.stat.exists|bool"
+
+- name: Ensure systemd reloaded after matrix-user-verification-service.service removal
+  service:
+    daemon_reload: yes
+  when: "matrix_user_verification_service_service_stat.stat.exists|bool"
+
+- name: Ensure Matrix user-verification-service paths don't exist
+  file:
+    path: "{{ matrix_user_verification_service_base_path }}"
+    state: absent
+
+- name: Ensure user-verification-service Docker image doesn't exist
+  docker_image:
+    name: "{{ matrix_user_verification_service_docker_image }}"
+    state: absent
```

Comment:
```suggestion
- name: Check existence of matrix-user-verification-service service
  stat:
    path: "{{ matrix_systemd_path }}/{{ matrix_user_verification_service_systemd_service_name }}"
  register: matrix_user_verification_service_service_stat

- when: matrix_user_verification_service_service_stat.stat.exists | bool
  block:
    - name: Ensure matrix-user-verification-service is stopped
      service:
        name: "{{ matrix_user_verification_service_systemd_service_basename }}"
        state: stopped
        daemon_reload: yes
      register: stopping_result

    - name: Ensure matrix-user-verification-service.service doesn't exist
      file:
        path: "{{ matrix_systemd_path }}/{{ matrix_user_verification_service_systemd_service_name }}"
        state: absent

    - name: Ensure Matrix user-verification-service paths don't exist
      file:
        path: "{{ matrix_user_verification_service_base_path }}"
        state: absent
```

Using a `block` helps with readability and we don't have to repeat the same `when` over and over again.

Reloading systemd is intentionally skipped (like it is in all / most other roles) - we only aim to reload during things like `--tags=start` in the systemd service manager role.

Deleting the Docker image is also skipped, because it's mostly useless (normally, it changes often.. and cleaning up just 1 out of 20 (for example) doesn't help much).